### PR TITLE
Board listing fixes

### LINKED ISSUE #69329: hwmv2: west boards output with identifiers contains board name as part of the identifier
~~It seems that `west boards --format "{identifiers}"` gives a python array output:~~
```
['stm32wb5mmg/stm32wb55xx']
['em_starterkit/emsk_em7d', 'em_starterkit/emsk_em9d', 'em_starterkit/emsk_em11d']
['emsdp/emsdp_em4', 'emsdp/emsdp_em5d', 'emsdp/emsdp_em6', 'emsdp/emsdp_em7d', 'emsdp/emsdp_em7d_esp', 'emsdp/emsdp_em9d', 'emsdp/emsdp_em11d']
['hsdk/arc_hsdk', 'hsdk/arc_hsdk/2cores']
['hsdk4xd/arc_hsdk4xd']
['iotdk/arc_iot']
['nsim/nsim_em', 'nsim/nsim_em7d_v22', 'nsim/nsim_em11d', 'nsim/nsim_hs', 'nsim/nsim_hs/sram', 'nsim/nsim_hs/smp', 'nsim/nsim_hs/flash_xip', 'nsim/nsim_hs/mpuv6', 'nsim/nsim_hs/hostlink', 'nsim/nsim_hs5x', 'nsim/nsim_hs5x/smp', 'nsim/nsim_hs5x/smp/12cores', 'nsim/nsim_hs6x', 'nsim/nsim_hs6x/smp', 'nsim/nsim_hs6x/smp/12cores', 'nsim/nsim_sem', 'nsim/nsim_sem/mpu_stack_guard', 'nsim/nsim_vpx5']
['robokit1/same70q21b']
['tlsr9518adk80d/tlsr9518']
```

~~Which does not go to stdout, it does not include any hwmv1 boards either~~

It seems that using the command `west boards --format "{identifiers}" | less` makes it look like there is no output, but there is output, but there is a huge amount of empty lines at the top (hwmv1 boards?) which should be fixed

### LINKED ISSUE #70614: Multiple samples are failing to build for the nrf52bsim in CI
**Describe the bug**
Multiple samples are failing to build for the nrf52bsim in CI

**To Reproduce**
1. mkdir build && cd build
1. cmake -GNinja  -DBOARD=nrf52_bsim  -DBOARD_ROOT=${ZEPHYR_BASE} ../samples/hello_world/
   OR
   cmake -GNinja  -DBOARD=native_sim  -DBOARD_ROOT=${ZEPHYR_BASE} ../samples/hello_world/
   OR 
   cmake -GNinja  -DBOARD=qemu_x86  -DBOARD_ROOT=${ZEPHYR_BASE} ../samples/hello_world/
2. See error

**Expected behavior**
No cmake errors

**Impact**
CI blocked for PRs triggering BT tests 

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/actions/runs/8391726266/job/22982731716?pr=70437#step:13:45
```
CMake Error at /__w/zephyr/zephyr/cmake/modules/boards.cmake:293 (message):
  Board qualifiers `` for board `nrf52_bsim` not found.  Please specify a
  valid board target.

  Valid board targets for are:

  nrf52_bsim/native

  NAME

  nrf52_bsim/nrf52_bsim

  nrf52_bsim/native
  ```

**Environment (please complete the following information):**
 - OS: Linux, Ubuntu 22.05
 - Toolchain: Host gcc 11/12
 - Commit SHA or Version used: Today's main e25c68be5dcced3ced2a6269df8e47c8921a8235

**Additional context**
* Introduced by 66b475a3aa4d73f4ddd5be5a7a1e0f3c7028e539
* Rerunning the same command in the same build folder, cmake succeeds 
* Removing the  `-DBOARD_ROOT` switch, cmake succeeds 
* Setting `-DBOARD_ROOT=..` , cmake succeeds

### PR DESCRIPTION
**Pull request by 57300**
Re-applies #69904 and re-fixes #70614

----

* Error out if multiple boards use the same name
* Ignore duplicate roots in list_boards/hardware

---

(edit by aescolar) Fixes: #69329

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
I would restore the original comment for the revert of the revert, maybe just mention that it's a second attempt in the message.

**Comment by 57300** [comment]
> I would restore the original comment for the revert of the revert, maybe just mention that it's a second attempt in the message.

I think that's reasonable. Replaced with a cherry-pick.

**Comment by 57300** [comment]
The bug observed in #70614 has two facets, both of which should now be fixed.

**Review by fabiobaltieri — APPROVED** [review]

File: cmake/modules/boards.cmake [hunk 1]
```diff
@@ -195,6 +195,24 @@ endif()
...
+
+      cmake_parse_arguments(LIST_BOARD "" "DIR" "" ${line})
+      set(board_dirs "${board_dirs}\n${LIST_BOARD_DIR}")
+
+      if(idx EQUAL -1)
+        break()
+      endif()
+      string(FIND "${ret_board}" "\n" idx REVERSE)
+    endwhile()
+    message(FATAL_ERROR "Multiple boards named '${BOARD}' found in:${board_dirs}")
```
Comment (tejlmand) [inline cmake/modules/boards.cmake:213]: I would prefer we detect this at the `list_boards.py` level, especially considering that in future we need to also support this: https://github.com/zephyrproject-rtos/zephyr/issues/69548

as well as ensuring that other commands, like `west boards` detect such errors as well.

However, as immediate fix, then this proposal is accepted and can be cleaned up when implementing https://github.com/zephyrproject-rtos/zephyr/issues/69548.

**Review by tejlmand — APPROVED** [review]

File: cmake/modules/boards.cmake (hunk 1 above)
Comment (57300) [reply cmake/modules/boards.cmake:213]: > I would prefer we detect this at the `list_boards.py` level

Agreed, and I'd like to get around to this as soon as I can.

I've already started work on refactoring the Python scripts, which would make this easier to detect at Python level, and would include performance improvements among other things.

**Review by nordicjm — APPROVED** [review]

**Pull request merged**